src/bun.js/bindings/BunProcess.cpp

```diff
@@ -3120,6 +3120,137 @@ JSC_DEFINE_HOST_FUNCTION(Process_functionCpuUsage, (JSC::JSGlobalObject * global
     RELEASE_AND_RETURN(throwScope, JSC::JSValue::encode(result));
 }
 
+JSC_DEFINE_HOST_FUNCTION(Process_functionThreadCpuUsage, (JSC::JSGlobalObject * globalObject, JSC::CallFrame* callFrame))
+{
+    auto& vm = JSC::getVM(globalObject);
+    auto throwScope = DECLARE_THROW_SCOPE(vm);
+
+#if OS(DARWIN)
+    mach_msg_type_number_t count;
+    thread_basic_info_data_t info;
+    kern_return_t kr;
+    thread_t thread;
+
+    thread = mach_thread_self();
+    count = THREAD_BASIC_INFO_COUNT;
+    kr = thread_info(thread,
+        THREAD_BASIC_INFO,
+        (thread_info_t)&info,
+        &count);
+
+    if (kr != KERN_SUCCESS) {
+        mach_port_deallocate(mach_task_self(), thread);
+        throwSystemError(throwScope, globalObject, "Failed to get thread CPU usage"_s, "thread_info"_s, kr);
+        return {};
+    }
+
+    auto* process = getProcessObject(globalObject, callFrame->thisValue());
+    Structure* cpuUsageStructure = process->cpuUsageStructure();
+
+    double user = std::chrono::microseconds::period::den * info.user_time.seconds + info.user_time.microseconds;
+    double system = std::chrono::microseconds::period::den * info.system_time.seconds + info.system_time.microseconds;
```

Comment:
Let's define a helper function:

```cpp
template<typename Sec, typename Usec>
double totalMicroseconds(Sec seconds, Usec microseconds)
{
    std::chrono::microseconds total = std::chrono::seconds(seconds) + std::chrono::microseconds(microseconds);
    return total.count();
}
```

Then change this to:

```suggestion
    double user = totalMicroseconds(info.user_time.seconds, info.user_time.microseconds);
    double system = totalMicroseconds(info.system_time.seconds, info.system_time.microseconds);
```

And so on for the other duration calculations.